src/codegen/llvm/codegen_llvm_visitor.cpp

```diff
@@ -20,6 +21,80 @@ namespace nmodl {
 namespace codegen {
 
 
+/****************************************************************************************/
+/*                            Helper routines                                           */
+/****************************************************************************************/
+
+
+void CodegenLLVMVisitor::visit_procedure_or_function(const ast::Block& node) {
+    const auto& name = node.get_node_name();
+    const auto& parameters = node.get_parameters();
+
+    // Procedure or function parameters are doubles by default.
+    std::vector<llvm::Type*> arg_types;
+    for (size_t i = 0, e = parameters.size(); i < e; ++i)
```

Comment:
just nitpick : should be fine to just do `i < parameters.size();` as well

Reply:
I think that's LLVM/MLIR style to cache upper bound. But not necessary, right!